src/models/BibDetails.ts

```diff
@@ -421,19 +408,49 @@ export default class BibDetails {
     }
   }
 
-  constructSubjectHeadingsArray(subject: string) {
+  constructSubjectLiteralsArray(subject: string) {
     // subject = "Italian food -- Spaghetti"
-    let stackedSubjectHeading = ""
+    let stackedSubjectLiteral = ""
 
     return subject
       .split(" -- ") // ["Italian food", "spaghetti"]
       .map((urlString, index) => {
         const dashDivided = index !== 0 ? " -- " : ""
         // First iteration "Italian food"
         // Second iteration "Italian food -- spaghetti"
-        stackedSubjectHeading = `${stackedSubjectHeading}${dashDivided}${urlString}`
+        stackedSubjectLiteral = `${stackedSubjectLiteral}${dashDivided}${urlString}`
 
-        return stackedSubjectHeading
+        return stackedSubjectLiteral
       })
   }
+
+  /**
+   * Flatten subject headings into a list of objects with a url and a label
+   */
+  flattenSubjectHeadingUrls(heading): BibDetailURL[] | null {
```

Comment:
is this an actual case you've come across? if so, in the future maybe this could return the corresponding value for a subjectLiteral